Detectors/TPC/reconstruction/macro/runCATracking.C

```diff
@@ -0,0 +1,81 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See https://alice-o2.web.cern.ch/ for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <vector>
+#include <fstream>
+#include <iostream>
+#include "TSystem.h"
+
+#include "TROOT.h"
+#include "TFile.h"
+#include "TTree.h"
+#include "TChain.h"
+#include "TClonesArray.h"
+
+#include "TPCBase/Defs.h"
+#include "TPCBase/CRU.h"
+#include "TPCBase/Sector.h"
+#include "TPCBase/Mapper.h"
+#include "TPCBase/PadRegionInfo.h"
+#include "TPCSimulation/Cluster.h"
+#include "TPCReconstruction/TPCCATracking.h"
+#include "TPCReconstruction/SyncPatternMonitor.h"
+#include "TPCReconstruction/TrackTPC.h"
+#include "DetectorsBase/Track.h"
+
+using namespace o2::TPC;
+using namespace std;
+
+void runCATracking(TString filename, TString outputFile, TString options, Int_t nmaxEvent=-1, Int_t startEvent=0) {
+  gSystem->Load("libTPCReconstruction.so");
+  TPCCATracking tracker;
+  vector<TrackTPC> tracks;
+  if (tracker.initialize(options.Data())) {
+    printf("Error initializing tracker\n");
+    return;
+  }
+
+  // ===| input chain initialisation |==========================================
+  TChain c("cbmsim");
+  c.AddFile(filename);
+
+  TClonesArray *clusters=0x0;
+  c.SetBranchAddress("TPCClusterHW", &clusters);
+
+  // ===| output tree |=========================================================
+  TFile fout(outputFile, "recreate");
+  TTree tout("events","events");
+  tout.Branch("Tracks", &tracks);
```

Comment:
I have a small doubt about this: The code does not seem to be using the usual FairTask / FairRootMgr mechanism for processing and IO. Is this supposed to be like this or just for prototyping purposes?

A future move to devices might be easier (automatic) if for now we stick to the FairTask processing.

Reply:
@sawenzel : in continuous mode detectors cannot rely on the FairRootMgr "event" segmentation obtained in the digitization and need to be able fetch the data on the pace they need. The same is true for the output: the granularity of the output chunk may have nothing to do with that of "input event" from the manager. An extra argument in favour of giving to reco. components an access to the input branch (alias of TF) rather than feeding single event is that some algorithms need to fill GPU buffers by as much data as possible before they start processing:  https://alice.its.cern.ch/jira/browse/AOC-43

Reply:
@shahor02 : Fair enough. But I would assume that the final granularity of clusters is anyway in terms if timeframes not of events. In any case, I understand this is currently for demonstrating purposes ...